fpsd/features.py

```diff
@@ -0,0 +1,768 @@
+                               "position <= {pos_stop} GROUP BY exampleid)     "
+                               "t{feat_ind} ON foo.exampleid =                 "
+                               "t{feat_ind}.exampleid").format(colname=feature_columns[x-1],
+                                                               pos_start=(x-1)*size_window,
+                                                               pos_stop=x*size_window,
+                                                               feat_ind=x)
+                              for x in range(1, num_features + 1)]
+
+            subqueries = " ".join(arr_subqueries)
+        else:
+            subqueries = ""
+
+        query = ("CREATE TABLE features.size_{}_windows AS ( "
+                 "SELECT foo.exampleid, {} FROM ( "
+                 "(SELECT exampleid, count(*) FROM " 
+                 "raw.frontpage_traces GROUP BY exampleid) foo"
+                 " {} ));").format(size_window,
+                               ', '.join(feature_columns),
+                               subqueries)
+
+        self.engine.execute(query)
+        return feature_table_name
+
+    def create_bursts(self):
+        """This method takes all examples and produces a table
+        public.current_bursts with all bursts in the following format:
+
+        burstid   | burst     | exampleid   | rank
+        (integer) | (bigint)  | (bigint)    | (bigint)
+        --------------------------------------------------
+        1         | 1         | 251         | 1
+        2         | 2         | 251         | 2
+        3         | 4         | 251         | 3
+
+        This table is then used by the burst table creation methods
+        called by generate_burst_tables().
+
+        Returns:
+            [string] name of newly created table
+        """
+
+        # Preprocessing that would ideally be done in SQL (for speed)
+        final_df = pd.DataFrame()
+        for example in tqdm(self.get_exampleids()):
+            trace_df = self.get_trace_cells(example)
+            bursts, ranks = compute_bursts(trace_df)
+            final_df = final_df.append(pd.DataFrame({'exampleid': example,
+                                                     'burst': bursts,
+                                                     'rank': ranks}))
+        final_df = final_df.reset_index().drop('index', axis=1)
+        self.drop_table("public.current_bursts")
+
+        table_creation = ("CREATE TABLE public.current_bursts             "
+                          "(burstid SERIAL PRIMARY KEY, exampleid BIGINT, "
+                          "burst BIGINT, rank BIGINT)                     ")
+        self.engine.execute(table_creation)
+
+        burst_rows = ['({}, {}, {})'.format(row[0], row[1], row[2])
+                      for row in final_df.values]
+
+        cols = final_df.columns
+        insert_query = ("INSERT INTO public.current_bursts "
+                        "({}, {}, {}) VALUES {};".format(cols[0],
```

Comment:
ya one can but it’s actually really slow to use pandas’ `DataFrame.to_sql` so I usually avoid it